core/src/main/java/io/github/tcdl/msb/collector/Collector.java

```diff
@@ -42,7 +41,8 @@
 
     private int timeoutMs;
     private int currentTimeoutMs;
-    private long waitForAcksUntil;
+    private Integer waitForAcksMs;
+    private Long waitForAcksUntil;
     private int waitForResponses;
```

Comment:
i think this need to be used in Requester also. but maybe i introduced a bug....